src/l500/ac-trigger.cpp

```diff
@@ -437,26 +446,107 @@ namespace ivcam2 {
             }
 #endif
 
-            auto dir = getenv( "RS2_DEBUG_DIR" );
-            if( dir )
+            std::string filename = get_debug_path_base();
+            if( ! filename.empty() )
             {
-                std::string filename = to_string()
-                    << dir
-                    << system_clock::now().time_since_epoch().count() * system_clock::period::num / system_clock::period::den
-                    << ".ac_log";
-
-                _f.open( filename );
-                if( _f  &&  _to_stdout )
-                    write_out( to_string() << "-D- CAH log is being written to: " << filename );
+                filename += ".ac_log";
+
+                _f_main.open( filename );
+                if( _f_main  &&  _to_stdout )
+                    write_out( to_string() << "-D- CAH main log is being written to: " << filename );
             }
 
-            if( _f || _to_stdout )
-                librealsense::log_to_callback( RS2_LOG_SEVERITY_ALL,
-                                               { this, []( rs2_log_callback * p ) {} } );
+            librealsense::log_to_callback( RS2_LOG_SEVERITY_ALL,
+                                            { this, []( rs2_log_callback * p ) {} } );
 
             AC_LOG( DEBUG, "LRS version: " << RS2_API_FULL_VERSION_STR );
         }
 
+        static void add_dir_sep( std::string & path )
+        {
+#ifdef _WIN32
+            char const dir_sep = '\\';
+#else
+            char const dir_sep = '/';
+#endif
+            if( ! path.empty() && path.back() != dir_sep )
+                path += dir_sep;
+        }
+
+        static std::string get_debug_path_base()
+        {
+            std::string path;
+
+            // If the user has this env var defined, then we write out logs and frames to it
+            // NOTE: The var should end with a directory separator \ or /
+            auto dir_ = getenv( "RS2_DEBUG_DIR" );
+            if( dir_ )
+            {
+                path = dir_;
+                add_dir_sep( path );
+
+                std::time_t now = std::time( nullptr );
```

Comment:
Consider reuse now_string() function. maybe with a default parameter of now_string(bool add_date = false)

Reply:
Good idea. Done.